pydicom/dataset.py

```diff
@@ -963,7 +963,9 @@ def _get_PIL_supported_compressed_pixeldata(self):
             generic_jpeg_file_header = (b'\xff\xd8\xff\xe0\x00\x10JFIF\x00',
                                         b'\x01\x01\x01\x00\x01\x00\x01\x00',
                                         b'\x00')
+            generic_jpeg_file_header = b''.join(generic_jpeg_file_header)
```

Comment:
This is not the problem. The problem is the way `generic_jpeg_file_header` is created. 
You should remove the comas
```py
            generic_jpeg_file_header = (b'\xff\xd8\xff\xe0\x00\x10JFIF\x00'
￼                                        b'\x01\x01\x01\x00\x01\x00\x01\x00'
￼                                        b'\x00')
```